File: 0118-pascals-triangle/0118-pascals-triangle.go
func generate(numRows int) [][]int {
    result := make([][]int, numRows)
    result[0] = make([]int, 1)
    result[0][0] = 1
    if numRows == 1 {
        return result
    }
    
    for i := 1; i < numRows; i++ {
        arr := make([]int, i + 1)
        
        for j := 0; j <= i; j++ {
            if j == 0 {
                arr[j] = 1
                continue
            }
            
            if j == i {
                arr[j] = 1
                continue
            }
            
            arr[j] = result[i - 1][j] + result[i - 1][j - 1]
        }
        
        result[i] = arr
    }
    
    return result
}

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();
        result.add(Arrays.asList(1));
        if (numRows == 1) {
            return result;
        }

        for (int i = 1; i < numRows; i++) {
            List<Integer> list = new ArrayList<>();

            for (int j = 0; j <= i; j++) {
                if (j == 0) {
                    list.add(1);
                    continue;
                }

                if (j == i) {
                    list.add(1);
                    continue;
                }

                list.add(result.get(i - 1).get(j) + result.get(i - 1).get(j - 1));
            }

            result.add(list);
        }

        return result;
    }
}